server/command.go

```diff
@@ -274,20 +275,18 @@ func (p *Plugin) SendLinksWithDetails(userID, channelID string, links []*storemo
 	sb.WriteString("| Mattermost Team | Mattermost Channel | MS Teams Team | MS Teams Channel | \n| :------|:--------|:-------|:-----------|")
 	errorsFound := false
 	wg := sync.WaitGroup{}
-	batchSize := 10
-	mattermostChannelIDsVsNames := make(map[string]string)
-	mattermostTeamIDsVsNames := make(map[string]string)
+	batchSize := math.Sqrt(float64(len(links)))
 
 	msTeamsTeamIDsVsNames := make(map[string]string)
 	msTeamsChannelIDsVsNames := make(map[string]string)
 	msTeamsTeamIDsVsChannelsQuery := make(map[string]string)
 
 	// Process all the links in batches using goroutines
-	for idx := 0; idx < len(links); idx += batchSize {
+	for idx := 0; idx < len(links); idx += int(batchSize) {
```

Comment:
This is still an unbounded number of goroutines. But I guess being limited to this places is good enough.

Reply:
Actually... I'm thinking. There is no longer IO inside this loop, I don't think we need batching or goroutines here. I would get rid of all that unneeded complexity and do it in a single loop.